CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+-------------------
+
+Patches are welcome!  We currently use the vanilla GitHub merge workflow, where
+individual pull requests are merged by a human via the web UI.
+
+During the review, any changes should be made as a new commit or an amended
+commit.  These can be force pushed to your branch and GitHub **should** be able
+to show the "interdiff" between pushes.
+
+Commit Message Style
+--------------------
+
+Please look at `git log` and match the commit log style, which is very
+similar to the
+[Linux kernel](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git).
+
+You may use `Signed-off-by`, but we're not requiring it.
+
+**General Commit Message Guidelines**:
+
+1. Title
+    - Specify the context or category of the changes e.g. `lib` for library changes, `docs` for document changes, `bin/<command-name>` for command changes, etc.
+    - Begin the title with the first letter of the first word capitalized.
+    - Aim for less than 50 characters, otherwise 72 characters max.
+    - Do not end the title with a period.
+    - Use an [imperative tone](https://en.wikipedia.org/wiki/Imperative_mood).
+2. Body
+    - Separate the body with a blank line after the title.
+    - Begin a paragraph with the first letter of the first word capitalized.
+    - Each paragraph should be formatted within 72 characters.
+    - Content should be about what was changed and why this change was made.
+    - If your commit fixes an issue, the commit message should end with `Closes: #<number>`.
+
+Commit Message example:
+
+```bash
+<context>: Less than 50 characters for subject title
+
+A paragraph of the body should be within 72 characters.
+
+This paragraph is also less than 72 characters.
+```
+
+For more information see [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+**Editing a Committed Message:**
+
+To edit the message from the most recent commit run `git commit --amend`. To change older commits on the branch use `git rebase -i`. For a successful rebase have the branch track `upstream master`. Once the changes have been made and saved, run `git push --force origin <branch-name>`.
+
+Testing Changes
+----------------
+
+At a bare minimum, your changes should pass the `tests/check.sh` script.  This is just
+some simple syntax checking, but is better than nothing at all.
+
+A more useful test of your changes would include the following:
+  - [ ] local build of `coreos-assembler` container image
+  - [ ] a build of `coreos-assembler` container image on [Quay.io](https://quay.io)
+  - [ ] a successful `coreos-assembler build && coreos-assembler run` of [Fedora CoreOS](https://github.com/coreos/fedora-coreos-config) using your changes in a "rootless" configuration (i.e. no use of `--privileged`)
+
+Code Style
+-----------
+
+We currently use a mix of `bash` shell scripts and Python.
+
+The current preference is to use spaces instead of tabs.  However, we will not start any holy wars over your own choice.
```

Comment:
Maybe add something that says "keep to the convention in the file if editing an existing file", "don't retab just to retab", and "use 4 spaces for python" (I'm assuming that last one isn't controversial given that's what python has ordained?

Reply:
In general I've seen the "keep to the convention in the file if editing an existing file" work as a good middle ground when a team can't reach consensus. The one problem I've seen with it is when people mistakenly add spaces to a "tabs" file and over time it gets a little more mixed. This happens both ways, but in my experience converting from tabs to spaces (i.e. fixing a spaces file that has mistakenly got tabs added to it) is way easier than converting from spaces to tabs.

Reply:
If there were some tool that helped us enforce this, it would be nice since I know I'm going to make a mistake (unintentionally).

Reply:
@ajeddeloh ^^ suggestions here

Reply:
another way to put this is:

- bob likes tabs
- alice likes spaces
- if bob makes a mistake, it's easy to fix
- if alice makes a mistake, it's not (not that I know of)

How can we make it easy to either prevent alice's mistake or fix later?

Reply:
I think its fine if we're not perfect, but trying to keep things the same is better than not trying at all, yes?

I think we should be more explicit that tabs or spaces is fine for bash, but tabs are _not_ ok for python (do we have agreement on this?) since the python language states that you should use spaces.